src/lib/device_tree/src/area.rs

```diff
@@ -15,6 +16,35 @@ pub struct Area {
     pub file: Option<String<U512>>,
 }
 
+pub fn get_kernel_area(dtfs_base: usize, dtfs_size: usize) -> Area {
+    let dtb = SectionReader::new(&Memory {}, dtfs_base, dtfs_size);
+    let areas = read_areas(&dtb).unwrap();
+    let mut last_area_end = 0;
+    let mut area_iter = areas.into_iter();
+
+    // Look for the first rompayload area and check for overlapping areas on the way
+    let area_opt = loop {
```

Comment:
Maybe add a comment that layoutflash ensures that fdt is sorted--otherwise the overlap check would be incomplete.

Reply:
Done.